channeldb/graph.go

```diff
@@ -3386,6 +3393,11 @@ func deserializeChanEdgePolicy(r io.Reader,
 	}
 	edge.FeeProportionalMillionths = lnwire.MilliSatoshi(n)
 
+	if err := binary.Read(r, byteOrder, &n); err != nil {
```

Comment:
This breaks any existing deployments as they won't have this new field serialized. Instead we either need a migration, or the any new fields added are to be at the end of the struct.

Reply:
Since the field data might already be present within the opaque data, we can either keep the current format, where we read and write the field to the var-bytes opaque data, or we can do a migration to transition it to a new format.

It is nice to avoid a migration, but reading and writing to the varbytes blob might be a bit cumbersome. I also think it might fail if the message flags indicate that the `max_htlc` field should be there, but it actually isn't, as we haven't validated the field before adding the policy to the DB.

So maybe a migration is cleanest?

Reply:
What was the rationale behind storing this field as `varbytes` btw? If we just always interpreted the last unknown bytes as the extra blob, I think it would be easier to let new fields "eat into" the extra bytes.